.github/workflows/build_docker_nightly.yml

```diff
@@ -12,6 +12,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          submodules: recursive
```

Comment:
This will pull all submodules, i.e. neurobagel-examples, bids-examples, and pipelines. Technically we only need the pipelines. It's only ~70MB (mainly for bids-examples), but that's ~ 1/3 of the whole image. Our other images are all a little large - the CLI is small by comparison. All that is to say: maybe it's worth to update the Dockerfile to not copy all the submodules in.